File: src/api/subsonic/albumInfo.ml
open Ppx_deriving_json_runtime.Primitives

type[@deriving json] albumInfo = {
  lastFmUrl : string option; [@json.option]
  musicBrainzId : string option; [@json.option]
  notes : string option; [@json.option]
}

type[@deriving json] body = { status : [ `ok ]; albumInfo : albumInfo }
type[@deriving json] t = body Response.t

File: src/api/subsonic/types/albumInfo.ts
import * as v from 'valibot';

export const AlbumInfo = v.object({
  lastFmUrl: v.optional(v.string()),
  musicBrainzId: v.optional(v.string()),
  notes: v.optional(v.string()),
});

export type AlbumInfo = v.InferInput<typeof AlbumInfo>;
